telemetry_core/src/main/java/com/newrelic/telemetry/SpanBatchSenderFactory.java

```diff
@@ -51,24 +38,18 @@ default SpanBatchSender createBatchSender(String apiKey, Duration callTimeout) {
    *     Relic API Keys</a>
    */
   default SenderConfigurationBuilder configureWith(String apiKey) {
-    return configureWith(apiKey, Duration.ofSeconds(2));
+    return SpanBatchSender.configurationBuilder().apiKey(apiKey).httpPoster(getPoster());
   }
 
+  HttpPoster getPoster();
+
   /**
-   * Create a new SpanBatchSenderBuilder with your New Relic Insights Insert API key and a custom
-   * http timeout.
+   * Create an {@link SpanBatchSenderFactory} with an HTTP implementation.
    *
-   * @see <a
-   *     href="https://docs.newrelic.com/docs/apis/getting-started/intro-apis/understand-new-relic-api-keys#user-api-key">New
-   *     Relic API Keys</a>
+   * @param creator A {@link Supplier} that returns an {@link HttpPoster} implementation.
```

Comment:
If the method's only parameter is of type `Supplier<HttpPoster>`, this javadoc reiterates the type and doesn't say where you might find one of these things.

Reply:
Please add this comment to #156